samples/samples_common.py

```diff
@@ -0,0 +1,29 @@
+# Common functions for sample code.
+
+import sys
+
+from getpass import getpass
+
+try:
+    get_input = raw_input
+except NameError:
+    get_input = input
+
+
+def get_user_details():
```

Comment:
It's probably fine in this case, especially since its only sample code, but generally its best to avoid using `sys.argv` outside the entry function/module and instead pass it in as a parameter. It makes it clearer to see what's going on, and what's actually parsing `sys.argv` and what's not.

i.e.: 

```python
def get_user_details(argv):
    ...

get_user_details(sys.argv)
```